From c1cd0d56e047a40afb2a59a56e1fd8043e448e05 Mon Sep 17 00:00:00 2001
From: Ryan VanderMeulen <rvandermeulen@mozilla.com>
Date: Tue, 4 Nov 2025 13:30:23 +0000
Subject: [PATCH] Bug 1998070 - Update macOS SDK to 26.1.
 r=releng-reviewers,firefox-build-system-reviewers,mac-reviewers,glandium,jcristau,mstange

Differential Revision: https://phabricator.services.mozilla.com/D271177
---
 build/moz.configure/toolchain.configure          |  2 +-
 .../configure/macos_fake_sdk/SDKSettings.plist   |  2 +-
 taskcluster/kinds/openh264-plugin/kind.yml       |  4 ++--
 taskcluster/kinds/toolchain/macos-sdk.yml        | 16 ++++++++--------
 taskcluster/kinds/toolchain/misc.yml             |  4 ++--
 taskcluster/scripts/misc/build-cpython.sh        |  2 +-
 taskcluster/scripts/misc/build-custom-car.sh     |  2 +-
 taskcluster/scripts/misc/build-geckodriver.sh    |  2 +-
 taskcluster/scripts/misc/build-gn-macosx.sh      |  2 +-
 taskcluster/scripts/misc/build-llvm-common.sh    |  4 ++--
 taskcluster/scripts/misc/build-nasm.sh           |  4 ++--
 taskcluster/scripts/misc/build-pkgconf.sh        |  4 ++--
 .../scripts/misc/build-rust-based-toolchain.sh   |  2 +-
 .../configs/openh264/macosx64-aarch64.py         |  4 ++--
 testing/mozharness/configs/openh264/macosx64.py  |  4 ++--
 15 files changed, 29 insertions(+), 29 deletions(-)

diff --git a/build/moz.configure/toolchain.configure b/build/moz.configure/toolchain.configure
index 99a95b9c645f5..3d8d2ea651fa1 100644
--- a/build/moz.configure/toolchain.configure
+++ b/build/moz.configure/toolchain.configure
@@ -233,7 +233,7 @@ with only_when(host_is_osx | target_is_osx):
     )
 
     def mac_sdk_min_version():
-        return "26.0"
+        return "26.1"
 
     @depends(
         "--with-macos-sdk",
diff --git a/python/mozbuild/mozbuild/test/configure/macos_fake_sdk/SDKSettings.plist b/python/mozbuild/mozbuild/test/configure/macos_fake_sdk/SDKSettings.plist
index 8b1792d034ab1..26bc190134c34 100644
--- a/python/mozbuild/mozbuild/test/configure/macos_fake_sdk/SDKSettings.plist
+++ b/python/mozbuild/mozbuild/test/configure/macos_fake_sdk/SDKSettings.plist
@@ -3,6 +3,6 @@
 <plist version="1.0">
 <dict>
         <key>Version</key>
-        <string>26.0</string>
+        <string>26.1</string>
 </dict>
 </plist>
diff --git a/taskcluster/kinds/openh264-plugin/kind.yml b/taskcluster/kinds/openh264-plugin/kind.yml
index 163051a911359..6f8b927e5ffc8 100644
--- a/taskcluster/kinds/openh264-plugin/kind.yml
+++ b/taskcluster/kinds/openh264-plugin/kind.yml
@@ -125,7 +125,7 @@ tasks:
                 - linux64-clang
                 - linux64-dump_syms
                 - linux64-nasm
-                - macosx64-sdk-26.0
+                - macosx64-sdk-26.1
     macosx64-aarch64/opt:
         attributes:
             build_platform: macosx64-aarch64
@@ -149,7 +149,7 @@ tasks:
                 - linux64-cctools-port
                 - linux64-clang
                 - linux64-dump_syms
-                - macosx64-sdk-26.0
+                - macosx64-sdk-26.1
     win32/opt:
         attributes:
             build_platform: win32
diff --git a/taskcluster/kinds/toolchain/macos-sdk.yml b/taskcluster/kinds/toolchain/macos-sdk.yml
index 34683fa9799ef..c1cab88a7d221 100644
--- a/taskcluster/kinds/toolchain/macos-sdk.yml
+++ b/taskcluster/kinds/toolchain/macos-sdk.yml
@@ -30,20 +30,20 @@ macosx64-sdk-13.3:
             - Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk
         toolchain-artifact: project/gecko/mac-sdk/MacOSX13.3.sdk.tar.zst
 
-macosx64-sdk-26.0:
-    description: "macOS 26.0 SDK"
+macosx64-sdk-26.1:
+    description: "macOS 26.1 SDK"
     treeherder:
-        symbol: TM(sdk26.0)
+        symbol: TM(sdk26.1)
     run:
         arguments:
-            - https://swcdn.apple.com/content/downloads/27/62/093-35114-A_AAH24ZZQB5/yn87ru9qe9225m8hwq2ic3hjy5yc5vw7h9/CLTools_macOSNMOS_SDK.pkg
-            - d3286bd6d4dff1b12f1d0dab4816719e605a1bfb76af84575deec37b25fc4462d4fc1258aa43b138a557c4f9304e06740441b101eb367a4b5bd77200b0708c71
-            - Library/Developer/CommandLineTools/SDKs/MacOSX26.0.sdk
-        toolchain-artifact: project/gecko/mac-sdk/MacOSX26.0.sdk.tar.zst
+            - https://swcdn.apple.com/content/downloads/22/09/093-00219-A_WIA1LP39TY/evbam2mb02xqr05ju9ddb95y8qil8kz9tm/CLTools_macOSNMOS_SDK.pkg
+            - f6a5f44b3652f5abdf7ad2602f54ad38774404f45264a4e70cfccc58b8b39aed2a89f74ef22c44bf932d0a55b062ebc3651f73fa06ff755ee5da53b9dcc62fba
+            - Library/Developer/CommandLineTools/SDKs/MacOSX26.1.sdk
+        toolchain-artifact: project/gecko/mac-sdk/MacOSX26.1.sdk.tar.zst
         toolchain-alias:
             - macosx64-sdk
             - macosx64-sdk-toolchain
-            - MacOSX26.0.sdk
+            - MacOSX26.1.sdk
 
 ios-sdk-18.4:
     description: "iPhoneOS18.4 SDK"
diff --git a/taskcluster/kinds/toolchain/misc.yml b/taskcluster/kinds/toolchain/misc.yml
index f3caaeb01bfc1..810e7b58659be 100644
--- a/taskcluster/kinds/toolchain/misc.yml
+++ b/taskcluster/kinds/toolchain/misc.yml
@@ -256,7 +256,7 @@ macosx-custom-car:
         toolchain-artifact: public/build/car_macosx.tar.zst
     fetches:
         toolchain:
-            - MacOSX26.0.sdk
+            - MacOSX26.1.sdk
 
 macosx-arm64-custom-car:
     description: "Custom chromium-as-release build for mac (apple silicon)"
@@ -292,7 +292,7 @@ macosx-arm64-custom-car:
         toolchain-artifact: public/build/car_macosx_arm.tar.zst
     fetches:
         toolchain:
-            - MacOSX26.0.sdk
+            - MacOSX26.1.sdk
 
 android-custom-car:
     description: "Custom chromium-as-release build for Android"
diff --git a/taskcluster/scripts/misc/build-cpython.sh b/taskcluster/scripts/misc/build-cpython.sh
index 4963134f3570a..62b5c42587e6d 100755
--- a/taskcluster/scripts/misc/build-cpython.sh
+++ b/taskcluster/scripts/misc/build-cpython.sh
@@ -37,7 +37,7 @@ case `uname -s` in
         esac
         # NOTE: both CFLAGS and CPPFLAGS need to be set here, otherwise
         # configure step fails.
-        sysroot_flags="-isysroot ${MOZ_FETCHES_DIR}/MacOSX26.0.sdk -mmacosx-version-min=${macosx_version_min}"
+        sysroot_flags="-isysroot ${MOZ_FETCHES_DIR}/MacOSX26.1.sdk -mmacosx-version-min=${macosx_version_min}"
         export CPPFLAGS="${sysroot_flags} -I${xz_prefix}/include"
         export CFLAGS=${sysroot_flags}
         export LDFLAGS="${LDFLAGS} ${sysroot_flags} -L${xz_prefix}/lib"
diff --git a/taskcluster/scripts/misc/build-custom-car.sh b/taskcluster/scripts/misc/build-custom-car.sh
index 7a1225a9f60d7..5ebc9d9116260 100755
--- a/taskcluster/scripts/misc/build-custom-car.sh
+++ b/taskcluster/scripts/misc/build-custom-car.sh
@@ -60,7 +60,7 @@ fi
 # Logic for macosx64
 if [[ $(uname -s) == "Darwin" ]]; then
   # Modify the config with fetched sdk path
-  export MACOS_SYSROOT="$MOZ_FETCHES_DIR/MacOSX26.0.sdk"
+  export MACOS_SYSROOT="$MOZ_FETCHES_DIR/MacOSX26.1.sdk"
   # Bug 1990712 & 1989676
   # HACK: Create a stub DarwinBasic.modulemap to satisfy Ninja’s dependency graph.
   # This file does not exist in Command Line Tools SDKs. It seems only the full
diff --git a/taskcluster/scripts/misc/build-geckodriver.sh b/taskcluster/scripts/misc/build-geckodriver.sh
index 89ffb4e457447..312643e5d6b01 100755
--- a/taskcluster/scripts/misc/build-geckodriver.sh
+++ b/taskcluster/scripts/misc/build-geckodriver.sh
@@ -25,7 +25,7 @@ case "$TARGET" in
 # OSX cross builds are a bit harder
 *-apple-darwin)
   export PATH="$MOZ_FETCHES_DIR/clang/bin:$PATH"
-  RUSTFLAGS="-Clinker=$MOZ_FETCHES_DIR/clang/bin/clang++ -C link-arg=-isysroot -C link-arg=$MOZ_FETCHES_DIR/MacOSX26.0.sdk -C link-arg=-fuse-ld=lld -C link-arg=--target=$TARGET"
+  RUSTFLAGS="-Clinker=$MOZ_FETCHES_DIR/clang/bin/clang++ -C link-arg=-isysroot -C link-arg=$MOZ_FETCHES_DIR/MacOSX26.1.sdk -C link-arg=-fuse-ld=lld -C link-arg=--target=$TARGET"
   if test "$TARGET" = "aarch64-apple-darwin"; then
       export MACOSX_DEPLOYMENT_TARGET=11.0
   else
diff --git a/taskcluster/scripts/misc/build-gn-macosx.sh b/taskcluster/scripts/misc/build-gn-macosx.sh
index ce61daa21d3a5..ca3f82f4c9336 100755
--- a/taskcluster/scripts/misc/build-gn-macosx.sh
+++ b/taskcluster/scripts/misc/build-gn-macosx.sh
@@ -5,7 +5,7 @@ set -e -v
 
 WORKSPACE=$HOME/workspace
 
-CROSS_SYSROOT=$MOZ_FETCHES_DIR/MacOSX26.0.sdk
+CROSS_SYSROOT=$MOZ_FETCHES_DIR/MacOSX26.1.sdk
 export MACOSX_DEPLOYMENT_TARGET=10.15
 
 export CC=$MOZ_FETCHES_DIR/clang/bin/clang
diff --git a/taskcluster/scripts/misc/build-llvm-common.sh b/taskcluster/scripts/misc/build-llvm-common.sh
index b96f68f437a97..75073e39b9962 100755
--- a/taskcluster/scripts/misc/build-llvm-common.sh
+++ b/taskcluster/scripts/misc/build-llvm-common.sh
@@ -39,11 +39,11 @@ case "$target" in
     -DCMAKE_LIPO=$MOZ_FETCHES_DIR/clang/bin/llvm-lipo
     -DCMAKE_SYSTEM_NAME=Darwin
     -DCMAKE_SYSTEM_VERSION=$MACOSX_DEPLOYMENT_TARGET
-    -DCMAKE_OSX_SYSROOT=$MOZ_FETCHES_DIR/MacOSX26.0.sdk
+    -DCMAKE_OSX_SYSROOT=$MOZ_FETCHES_DIR/MacOSX26.1.sdk
     -DCMAKE_EXE_LINKER_FLAGS=-fuse-ld=lld
     -DCMAKE_SHARED_LINKER_FLAGS=-fuse-ld=lld
     -DDARWIN_osx_ARCHS=$arch
-    -DDARWIN_osx_SYSROOT=$MOZ_FETCHES_DIR/MacOSX26.0.sdk
+    -DDARWIN_osx_SYSROOT=$MOZ_FETCHES_DIR/MacOSX26.1.sdk
     -DDARWIN_macosx_OVERRIDE_SDK_VERSION=11.0
     -DDARWIN_osx_BUILTIN_ARCHS=$arch
     -DLLVM_DEFAULT_TARGET_TRIPLE=$target
diff --git a/taskcluster/scripts/misc/build-nasm.sh b/taskcluster/scripts/misc/build-nasm.sh
index 8ee37fba4988b..5490ca946eaaf 100755
--- a/taskcluster/scripts/misc/build-nasm.sh
+++ b/taskcluster/scripts/misc/build-nasm.sh
@@ -38,13 +38,13 @@ case "$1" in
     macosx64)
         export MACOSX_DEPLOYMENT_TARGET=10.15
         TARGET=x86_64-apple-darwin
-        CC="clang -fuse-ld=lld --target=$TARGET -isysroot $MOZ_FETCHES_DIR/MacOSX26.0.sdk"
+        CC="clang -fuse-ld=lld --target=$TARGET -isysroot $MOZ_FETCHES_DIR/MacOSX26.1.sdk"
         EXE=
 	;;
     macosx64-aarch64)
         export MACOSX_DEPLOYMENT_TARGET=11.0
         TARGET=aarch64-apple-darwin
-        CC="clang -fuse-ld=lld --target=$TARGET -isysroot $MOZ_FETCHES_DIR/MacOSX26.0.sdk"
+        CC="clang -fuse-ld=lld --target=$TARGET -isysroot $MOZ_FETCHES_DIR/MacOSX26.1.sdk"
         EXE=
 	;;
     *)
diff --git a/taskcluster/scripts/misc/build-pkgconf.sh b/taskcluster/scripts/misc/build-pkgconf.sh
index 3c40c07a2a390..805fbd3d4bd59 100755
--- a/taskcluster/scripts/misc/build-pkgconf.sh
+++ b/taskcluster/scripts/misc/build-pkgconf.sh
@@ -21,13 +21,13 @@ aarch64-unknown-linux-gnu)
 x86_64-apple-darwin)
     export MACOSX_DEPLOYMENT_TARGET=10.15
     TARGET=$1
-    CC="clang --target=$TARGET -isysroot $MOZ_FETCHES_DIR/MacOSX26.0.sdk"
+    CC="clang --target=$TARGET -isysroot $MOZ_FETCHES_DIR/MacOSX26.1.sdk"
     EXE=
     ;;
 aarch64-apple-darwin)
     export MACOSX_DEPLOYMENT_TARGET=11.0
     TARGET=$1
-    CC="clang --target=$TARGET -isysroot $MOZ_FETCHES_DIR/MacOSX26.0.sdk"
+    CC="clang --target=$TARGET -isysroot $MOZ_FETCHES_DIR/MacOSX26.1.sdk"
     EXE=
     ;;
 x86_64-pc-windows-gnu)
diff --git a/taskcluster/scripts/misc/build-rust-based-toolchain.sh b/taskcluster/scripts/misc/build-rust-based-toolchain.sh
index 6e0fb418ccb3a..b6091a3b34f87 100755
--- a/taskcluster/scripts/misc/build-rust-based-toolchain.sh
+++ b/taskcluster/scripts/misc/build-rust-based-toolchain.sh
@@ -42,7 +42,7 @@ aarch64-unknown-linux-gnu)
     else
         export MACOSX_DEPLOYMENT_TARGET=10.15
     fi
-    MACOS_SYSROOT=$MOZ_FETCHES_DIR/MacOSX26.0.sdk
+    MACOS_SYSROOT=$MOZ_FETCHES_DIR/MacOSX26.1.sdk
     export RUSTFLAGS="-Clinker=$MOZ_FETCHES_DIR/clang/bin/clang++ -C link-arg=-isysroot -C link-arg=$MACOS_SYSROOT -C link-arg=-fuse-ld=lld -C link-arg=--target=$TARGET $rust_lto_flags"
     export CC="$MOZ_FETCHES_DIR/clang/bin/clang"
     export CXX="$MOZ_FETCHES_DIR/clang/bin/clang++"
diff --git a/testing/mozharness/configs/openh264/macosx64-aarch64.py b/testing/mozharness/configs/openh264/macosx64-aarch64.py
index 94a0a3f914cd3..081c3e387ba85 100644
--- a/testing/mozharness/configs/openh264/macosx64-aarch64.py
+++ b/testing/mozharness/configs/openh264/macosx64-aarch64.py
@@ -22,14 +22,14 @@
     "partial_env": {
         "CFLAGS": (
             "-target aarch64-apple-darwin -mcpu=apple-a12 "
-            "-isysroot {MOZ_FETCHES_DIR}/MacOSX26.0.sdk "
+            "-isysroot {MOZ_FETCHES_DIR}/MacOSX26.1.sdk "
             "-mmacosx-version-min=11.0".format(
                 MOZ_FETCHES_DIR=os.environ["MOZ_FETCHES_DIR"]
             )
         ),
         "LDFLAGS": (
             "-target aarch64-apple-darwin -mcpu=apple-a12 "
-            "-isysroot {MOZ_FETCHES_DIR}/MacOSX26.0.sdk "
+            "-isysroot {MOZ_FETCHES_DIR}/MacOSX26.1.sdk "
             "-mmacosx-version-min=11.0".format(
                 MOZ_FETCHES_DIR=os.environ["MOZ_FETCHES_DIR"]
             )
diff --git a/testing/mozharness/configs/openh264/macosx64.py b/testing/mozharness/configs/openh264/macosx64.py
index e01d739763439..bd76dfb247927 100644
--- a/testing/mozharness/configs/openh264/macosx64.py
+++ b/testing/mozharness/configs/openh264/macosx64.py
@@ -22,14 +22,14 @@
     "partial_env": {
         "CXXFLAGS": (
             "-target x86_64-apple-darwin "
-            "-isysroot {MOZ_FETCHES_DIR}/MacOSX26.0.sdk "
+            "-isysroot {MOZ_FETCHES_DIR}/MacOSX26.1.sdk "
             "-mmacosx-version-min=10.12".format(
                 MOZ_FETCHES_DIR=os.environ["MOZ_FETCHES_DIR"]
             )
         ),
         "LDFLAGS": (
             "-target x86_64-apple-darwin "
-            "-isysroot {MOZ_FETCHES_DIR}/MacOSX26.0.sdk "
+            "-isysroot {MOZ_FETCHES_DIR}/MacOSX26.1.sdk "
             "-mmacosx-version-min=10.12".format(
                 MOZ_FETCHES_DIR=os.environ["MOZ_FETCHES_DIR"]
             )
